etcd/etcd.go

```diff
@@ -167,6 +170,15 @@ func (c *Client) tryEndpoints(mreq func(string) (*http.Request, error)) (*http.R
 			break
 		}
 
+		if isTimeout(err) {
+			break
+		} else if uerr, ok := err.(*url.Error); ok {
+			if isTimeout(uerr.Err) {
+				err = uerr.Err
+				break
+			}
+		}
+
```

Comment:
I am probably missing something here. It seems to me that a timeout on the first endpoint prevents you from trying any remaining endpoint.